### ISSUE TITLE
getting MS TEAMS meeting invites that do not show time and date of meeting

### ISSUE THREAD

**Issue by gstlouisgit** [webmin/webmin#2160]
I have clients that use webmin, beautiful UI and great webmail.  I came across a problem, which I believe is known on some webmail apps.  [https://answers.microsoft.com/en-us/msteams/forum/all/microsoft-teams-meeting-invites-not-showing-date/bbf9eadc-1f44-4876-8678-5f985c94950b](url)

If you ensure you are seeing the email in HTML in webmail of the Microsoft Teams Meeting Invite, you can see the links etc, but there is no date and time of the meeting invite.  If I forward this same email in thunderbird let's say, you can see the invite details.  I'm assuming this is because thunderbird picks up the attach ICS file for its details?

Anyway, in webmin I looked into an option to try to get this working and I think there just isn't.  Here is how to email looks like.  I don't know if there is a way in webmin to be able to see this?

![Screen Shot 2024-05-17 at 8 38 40 AM](https://github.com/webmin/webmin/assets/20375753/ee4dfe21-e7bc-496e-bab7-c86bac3bcf0c)

**Comment by iliajie** [comment]
Hello,

> If you ensure you are seeing the email in HTML in webmail of the Microsoft Teams Meeting Invite, you can see the links etc, but there is no date and time of the meeting invite.

This is a bit hard to follow. Could you share an email (original source file) that isn't being displayed as expected and a screenshot of how it should look? You can send it to my email address (which you can deduct by visiting my GitHub user profile), and I'll take a closer look.

**Comment by gstlouisgit** [comment]
Hello @iliajie
I cannot seem to find your email in your profile.  Can you write to my email so I have it?
Have a great weekend!

**Comment by iliajie** [comment]
> I cannot seem to find your email in your profile.

You need to guess my email based on the already provided domain name. It should be very obvious.

**Comment by chris001** [comment]
@gstlouisgit I'd try firstname at webmin dot com

**Comment by iliajie** [comment]
> @gstlouisgit I'd try firstname at webmin dot com

This is not what I suggested. --) However, webmin dot dev should work.

**Comment by chris001** [comment]
Thanks.  firstname at ross dot gdn might also work!

**Comment by iliajie** [comment]
Thanks @gstlouis for contacting me! I have taken a closer look, and one improvement we could make in Webmin and Usermin is to support displaying calendar events directly in the message, if an email has an `.ics` file attached.

**Comment by iliajie** [comment]
Alright, I've taken a closer look and implemented a custom parser for iCalendar files in [this PR](https://github.com/webmin/webmin/pull/2189). This will allow us to generate an event HTML table and insert it into the email when displaying it in the UI.

**Comment by chris001** [comment]
Very cool! 😎

**Comment by iliajie** [comment]
This feature has been implemented! To be clear, it took **many, many hours** to accomplish! I really hope you all are going to give it a good test and happily use it ever after!

This is how the email with any iCalendar event will look like in Webmin and Usermin email, e.g.:

<img width="1224" alt="image" src="https://github.com/webmin/webmin/assets/4426533/6c667ed7-677f-4afb-845b-9f4bb813f2dd">

The above displays basic details about the event, though clicking the arrow under **Who** field will display more details about the event, e.g.:

<img width="1222" alt="image" src="https://github.com/webmin/webmin/assets/4426533/11898c0c-503f-452a-9e30-c55ac766dc93">

**Comment by gstlouisgit** [comment]
that is just beautiful.  I have several clients using usermin which will be testing this out.  Thank you again for spending the time.  This will be a great addition if usermin has caldav/cardav to accept these appointments!

**Comment by iliajie** [comment]
I have fully integrated embedded calendars into Webmin and Usermin emails, ensuring consistent styling across both themes.

For those interested in trying it out immediately, it's available in the nightly builds of Webmin and Usermin.
